app/models/user.rb

```diff
@@ -13,7 +13,7 @@ def has_membership_application?
 
 
   def has_company?
-    has_membership_application? && (membership_applications.select { |app| app.company }).count > 0
+    has_membership_application? && (membership_applications.includes(:company).select { |app| app.company }).count > 0
```

Comment:
I suspect that this might be more efficient:
```
def has_company?
  membership_applications.where.not(company_id: nil).count > 0
end

This avoids loading all MA objects and instantiating an array.
```

Reply:
FYI I ran a quick benchmark:
```
require 'benchmark'

Benchmark.bmbm do |x|
  x.report('faster') do
    100.times do
      User.find(2).membership_applications.where.not(company_id: nil).count > 0
    end
  end

  x.report('slower') do
    100.times do
      user = User.find(2)
      user.membership_applications.size > 0 &&
        (user.membership_applications.includes(:company).select { |app| app.company }).count > 0
      end
    end
  end
end
```
and got these results:
```
Rehearsal ------------------------------------------
faster   0.100000   0.010000   0.110000 (  0.149526)
slower   0.290000   0.020000   0.310000 (  0.393786)
--------------------------------- total: 0.420000sec

             user     system      total        real
faster   0.100000   0.000000   0.100000 (  0.136219)
slower   0.240000   0.010000   0.250000 (  0.300988)
```
COU time is low for both.  The difference it probably swapping things into memory (and perhaps GC).

Reply:
You're absolutely right.

Just for kicks, I've tested the same setup for a user with 100 membership applications. Although a mostly academic situation, it drives home the point even more ;)
```
Rehearsal ------------------------------------------
faster   0.160000   0.020000   0.180000 (  0.271643)
slower   1.490000   0.050000   1.540000 (  1.685112)
--------------------------------- total: 1.720000sec

             user     system      total        real
faster   0.110000   0.010000   0.120000 (  0.154220)
slower   1.320000   0.020000   1.340000 (  1.431642)

```